_posts/2016-09-10-akka-stream-kafka.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: series_post
+title: "Akka Streams Kafka 0.11"
+description: ""
+author: Adam Warski
+category: integrations
+series_title: Integration
+series_tag: integration
+tags: [streams,integration]
+---
+{% include JB/setup %}
+
+
+[Apache Kafka](http://kafka.apache.org) is the leading distributed messaging system, and [Reactive Streams](http://www.reactive-streams.org) is an emerging standard for asynchronous stream processing. It seems natural to combine these two; that’s why [SoftwareMill](https://softwaremill.com) started the [reactive-kafka](https://github.com/akka/reactive-kafka) project back in December 2014 and maintained it since.
+
+Recently we’ve combined efforts with the Akka team, and now the project can leverage the expert input from the [Lightbend](https://www.lightbend.com) Akka maintainers themselves (Patrik Nordwall and Endre Varga), community ([Alexey Romanchuk](https://github.com/13h3r)) and the original reactive-kafka authors (Krzysiek Ciesielski from [Softwaremill](https://softwaremill.com)).
+
+As a result of this collaboration, we’re pleased to announce the 0.11 release of the project, which brings a lot of updates! Integrating Kafka into your reactive data processing pipeline is now even easier than before.
+
+First of all, we have a redesigned API, which should be in-line with other reactive-* connectors (see also the [Alpakka](http://blog.akka.io/integrations/2016/08/23/intro-alpakka) initiative), as well as provide additional flexibility. As an example of a common task, to read data from a Kafka topic, process it and commit offsets in batches collected within 10 seconds or at most 30 messages, you can define your stream as follows:
+
+```scala
+val consumerSettings = ConsumerSettings(system, 
+  new ByteArrayDeserializer, new StringDeserializer)
+  .withBootstrapServers("localhost:9092")
+  .withGroupId("group1")
+  .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+
+val result = Consumer
+  .committableSource(consumerSettings, Subscriptions.topics("topic1"))
+  .mapAsync(3)(processMsg)
+  .groupedWithin(30, 10.seconds) // commit every 10 seconds or 30 msgs
+  .map(group => group.foldLeft(CommittableOffsetBatch.empty) { 
+    (batch, elem) => batch.updated(elem) })
+  .mapAsync(3)(_.commitScaladsl())
+  .runWith(Sink.ignore)
+
+result.onFailure {
+  case e: Throwable => handleError(e)
+}
+```
+
+Secondly, there are significant performance improvements. We’ve done some benchmarks, and while there is still some overhead from the reactive-kafka wrapper (though keep in mind, that you also get something in return: e.g. all of the benefits coming from backpressure!), the overall numbers look very good and are getting better.
```

Comment:
Can we add that it also is a great API, much easier to work with than the native kafka one?